ACPT-1493: Reset via reflection for ApplicationServer (#286)
AdobeDocs/commerce-php: src/pages/development/components/app-server.md

```php
class Cache implements ResetAfterRequestInterface
```

> New in 2.4.8: ResetAfterRequestInterface doesn't have to be added to the class. The `_resetState()` method will be found by reflection and called as if it was implementing ResetAfterRequestInterface.  This feature was added to allow modules to be backwards compatible with previous versions < 2.4.7 which don't have this interface.

Reviewer: ```suggestion
In the `2.4.8-beta`, you do not have to add `ResetAfterRequestInterface` to the class. The `_resetState()` method is found by reflection and called by implementing `ResetAfterRequestInterface`.  This feature allows modules to be backwards compatible with previous versions before `2.4.7` that do not have this interface.
```

    $this->data = [];
}
```

> 
> New in 2.4.8: An optional alternative to adding a _resetState method to a class has been added. Instead of adding _resetState to a class, a reset.json file may be added to a modules etc directory.  The reset.json file will define which properties of a class will be reset, and what they will be reset to after a request.

Reviewer: ```suggestion
<InlineAlert variant="info" slots="text"/>

The following section only applies to Adobe Commerce `2.4.8-beta`:

The `2.4.8-beta` includes an alternative to the `a _resetState` method. Instead of adding `_resetState` to a class, you can add a `reset.json` file to a module's `etc` directory.  The `reset.json` file defines which class properties are reset, and what they reset to after a request.
```

> This feature was added to allow modules to be backwards compatible with previous versions < 2.4.7 which don't have the ResetAfterRequestInterface interface and also for classes where new public functions cannot be added.  It also makes it possible to add reset behaviour to classes for modules that aren't under your control.
> 
> Here's an example. A class `Magento\Reward\Model\Total\Quote\Reward` inherits from `Magento\Quote\Model\Quote\Address\Total\AbstractTotal`.  A reset.json file could be added to both of their modules.

Reviewer: ```suggestion
This feature allows modules to be backwards compatible with previous versions `2.4.7` and earlier, which do not have the `ResetAfterRequestInterface` interface. This feature is also compatible with classes that do not allow adding new public functions. You can also add reset behavior to classes for modules that you do not control.

Here's an example. A class `Magento\Reward\Model\Total\Quote\Reward` inherits from `Magento\Quote\Model\Quote\Address\Total\AbstractTotal`.  A `reset.json` file could be added to both modules:
```

> ```

Reviewer: ```suggestion
```PHP
```

"Magento\\Quote\\Model\\Quote\\Address\\Total\\AbstractTotal": {
  "_code": null,
  "_address": null,
  "_canAddAmountToAddress": true,
  "_canSetAddressAmount": true,
  "_itemRowTotalKey": null,
  "total": null
}
```

and

> ```

Reviewer: ```suggestion
```PHP
```

{
  "Magento\\Reward\\Model\\Total\\Quote\\Reward": {
    "_code": "_code"
  }
}
```

> Both of these reset.json definitions will be used for a Reward object, and they will be called in proper order of inheritance so that subclasses can have specializations to their reset values that get called after the base class.

Reviewer: ```suggestion
Both `reset.json` definitions are used for a Reward object. They are called in proper order of inheritance, so subclasses can have specializations to their reset values that get called after the base class.
```
